modules/machineset-customer-managed-encryption-azure.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc
+
+[id="machineset-enabling-customer-managed-encryption-azure_{context}"]
+= Enabling customer-managed encryption keys for a machine set
+
+Azure allows users to supply an encryption key to encrypt data on managed disks at rest. You can enable server-side encryption with customer-managed keys by using the Machine API.
+
+An Azure Key Vault, a disk encryption set, and an encryption key are required to use a customer-managed key. The disk encryption set must preside in a resource group where the Cloud Credential Operator (CCO) has granted permissions. If not, an additional reader role is required to be granted on the disk encryption set.
+
+.Prerequisites
+
+* link:https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys#create-an-azure-key-vault-instance[Create an Azure Key Vault instance].
+* link:https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys#create-an-instance-of-a-diskencryptionset[Create an instance of a disk encryption set].
+* link:https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys#grant-the-diskencryptionset-access-to-key-vault[Grant the disk encryption set access to key vault].
+
+.Procedure
+
+. Configure the disk encryption set under the `providerSpec` field in your machine set YAML file. For example:
```

Comment:
If we can expect readers to understand this schema pretty well if they're reading this module, this is cool. If they might not know it too well, we might consider adding some more info here w/callouts.